_rules/inline-link-in-paragraph-is-distinguishable-be4d0c.md

```diff
@@ -0,0 +1,383 @@
+---
+id: be4d0c
+name: Inline link is distinguishable
+rule_type: composite
+description: |
+  This rule checks that inline links have a way to distinguish them from any surrounding text other than color alone.
+accessibility_requirements: # Remove whatever is not applicable
+  wcag20:1.4.1: # Use of Color (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_rules:
+  - e7cc2b
+  - 88407d
+  - 36f116
+  - 66e9f0
+  - 548868
+  - 2803b8
+acknowledgments:
+  authors:
+    - Brian Bors
+    - Carlos Duarte
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to any [visible][] HTML element that is a [semantic link][], for which all the following is true:
+
+- **link text**: The element has [visible][] [text nodes][text node] as [descendants][descendant] in the [flat tree][]; and
+- **non-link line text**: The element is [rendered on a line][] containing [visible][] [text nodes][text node] that are not [descendants][descendant] in the [flat tree][] of a [semantic link][].
+
+## Expectation
+
+For each test target, the [outcome][] of one of the following rules is "passed":
+
+- [Inline link has distinguishable content](https://act-rules.github.io/rules/e7cc2b)
+- [Inline link has distinguishable style not based on color alone](https://act-rules.github.io/rules/88407d)
+- [Inline link has distinguishable border](https://act-rules.github.io/rules/36f116)
+- [Inline link has distinguishable box-shadow](https://act-rules.github.io/rules/66e9f0)
+- [Inline link has different foreground color and distinguishable style on hover and focus](https://act-rules.github.io/rules/548868)
+- [Inline link has different background color and distinguishable style on hover and focus](https://act-rules.github.io/rules/2803b8)
+
+## Assumptions
```

Comment:
You need an assumption that HTML has not been mishandled too heavily.

I can have a link without styling at one end of the DOM, and an element at the other end of the DOM which has a border and happen to be rendered just on top of the link due to some "clever" positioning. This would fail the rule (the link has no distinguishable styling or content) but pass the SC (the link is totally visually different from the rest).

Even worse, it could be a distant `canvas` which happens to draw a rectangle just around the link, so no style is applied.

Reply:
Added an assumption for this.